modin/pandas/test/dataframe/test_protocol.py

```diff
@@ -0,0 +1,244 @@
+import modin.pandas as pd
+import numpy as np
+import pytest
+from modin.pandas.test.utils import df_equals
+from modin.core.dataframe.pandas.exchange.dataframe_protocol.utils import from_dataframe
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {"a": [1.5, 2.5, 3.5], "b": [9.2, 10.5, 11.8]},
+        {"a": [1, 2, 3], "b": [3, 4, 5], "c": [1.5, 2.5, 3.5], "d": [9, 10, 11]},
+    ],
+)
+def test_float_only(data):
+    df = pd.DataFrame(data)
+    df_equals(df, from_dataframe(df.__dataframe__()))
+
+
+def test_noncontiguous_columns():
+    arr = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+    df = pd.DataFrame(arr, columns=["a", "b", "c"])
+    assert df["a"].to_numpy().strides == (24,)
```

Comment:
Can we rely on this strides assertion? I get different values running on different machines (Windows and Linux):
<details><summary>Reproducer</summary>

```python
import modin.pandas as pd
import numpy as np
import modin.config as cfg
from modin import set_execution


cfg.IsExperimental.put(False)
set_execution("Ray", "Pandas")

arr = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
df = pd.DataFrame(arr, columns=["a", "b", "c"])
print("PandasOnRay: ", df["a"].to_numpy().strides)

import pandas
df = pandas.DataFrame(arr, columns=["a", "b", "c"])
print("Pure pandas: ", df["a"].to_numpy().strides)
```

</details>

Windows output:
```
PandasOnRay:  (4,)
Pure pandas:  (12,)
```
Linux output:
```
PandasOnRay:  (8,)
Pure pandas:  (24,)
```

Reply:
As far as I understand, the original strides assertion was taken from pandas tests and was intended to verify that the columns ARE NOT stored contiguously in memory (`df["a"].to_numpy().strides == (24,)` means that offset between the elements is 24 bytes, or 3 int64 values). Windows use `int32` by default and linux `int64`, that's why this assertion is OS dependent, we can forcibly pin dtypes to avoid it I guess.

BTW, the current assertion verifies that the columns ARE stored contiguous as the offset is equal to the number of bytes for used integer type, this happens because `df[col].to_numpy` aparently copies the data into continguous buffer, so this assertion makes no sense. We can verify that the data is stored by strides internally by doing something like:
```python
assert df._query_compiler._modin_frame._partitions[0][0].to_pandas()["a"].to_numpy().strides == (24, 8)
```

cc @vnlitvinov

Reply:
@dchigarev but this isn't correct.

assert df._query_compiler._modin_frame._partitions[0][0].to_pandas()["a"].to_numpy().strides == (24, 8)
E       assert (8,) == (24, 8)
E         At index 0 diff: 8 != 24
E         Right contains one more item: 8
E         Use -v to get the full diff

Reply:
Meaning that when the frame is constructed using high level `modin.pandas.DataFrame` we probably can't ensure whether the data is stored by strides inside partitions (it probably does a copy of the passed data somewhere).

@vnlitvinov do we need this test at all then?